pkg/plugins/pg/destination/destination.go

```diff
@@ -16,114 +16,178 @@ package destination
 
 import (
 	"context"
-	"database/sql"
+	"encoding/json"
 	"fmt"
 	"strings"
 
-	"github.com/conduitio/conduit/pkg/foundation/cerrors"
-	"github.com/conduitio/conduit/pkg/plugins"
-	"github.com/conduitio/conduit/pkg/record"
+	"github.com/conduitio/conduit/pkg/plugin/sdk"
 
 	sq "github.com/Masterminds/squirrel"
-	_ "github.com/lib/pq" // Blank import to register the PostgreSQL driver
+	"github.com/jackc/pgx/v4"
 )
 
-// Destination fulfills the Destination interface
-type Destination struct {
-	Position record.Position
+// Postgres requires use of a different variable placeholder.
+var psql = sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
 
-	db *sql.DB
+type Adapter struct {
+	sdk.UnimplementedDestination
+
+	Position sdk.Position
+
+	conn   *pgx.Conn
+	config config
 }
 
-// Open attempts to open a database connection to Postgres.
-func (s *Destination) Open(ctx context.Context, cfg plugins.Config) error {
-	url, ok := cfg.Settings["url"]
-	if !ok {
-		return cerrors.New("must provide a connection URL")
+const (
+	actionDelete = "delete"
+	actionInsert = "insert"
+	actionUpdate = "update"
+)
+
+type config struct {
+	url           string
+	tableName     string
+	keyColumnName string
+}
+
+func New() sdk.Destination {
+	return &Adapter{}
+}
+
+func (d *Adapter) Configure(ctx context.Context, cfg map[string]string) error {
```

Comment:
I'd rather rename Adapter -> Destination.

Reply:
I too like Destination more (it was used before too).

Reply:
I changed it to avoid the stutter (destination.Destination) but I can update it to Destination 👍

Reply:
> I changed it to avoid the stutter (destination.Destination)

That's why advocate to not use the `source` and `destination` packages in future connectors and in the connector template (at least not for the structs that implement the `sdk` interfaces).